File: do-not-forget-frontend/src/main/resources/webapp/src/model/Task.ts
import {TaskState} from "./TaskState";
import {CustomDate} from "./CustomDate";
import {PriorityProvider} from "../providers/priority.provider";
export class Task {
    id: string;
    estimateTime: number = 0;
    spendTime: number = 0;
    deadLine: CustomDate = {day: new Date().getDate(), month: new Date().getMonth() + 1, year: new Date().getFullYear()};
    addedDate: CustomDate = new CustomDate;
    priority: number = new PriorityProvider().retrieveDefaultPriorityIndex();
    title: string = 'Something to do..';
    description: string = '';
    state: string = TaskState.NEW;
}

File: do-not-forget-backend/src/main/java/pl/pw/as/model/task/Task.java
package pl.pw.as.model.task;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Task {
    @Id
    private String id;
    private int estimateTime;
    private int spendTime;
    private int priority;
    private LocalDate deadLine;
    private LocalDate addedDate;
    private String title;
    private String description;
    private TaskState state;

}
